[android] No matching client found for package name 'app.organicmaps.beta'

```
 ./gradlew -Pfirebase=false assembleGoogleBeta

> Configure project :app
Building with Google Mobile Services
Building without Google Firebase Services
Building with Android Auto
Version: 2023.11.24-8
VersionCode: 23112408
Building for [armeabi-v7a, arm64-v8a, x86_64] archs.
Create separate apks: false

> Task :app:processGoogleBetaGoogleServices FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processGoogleBetaGoogleServices'.
> No matching client found for package name 'app.organicmaps.beta'
```

Why does -Pfirebase=false still require `android/app/google-services.json`? Beta should work out of the box without that file.

--- a/android/app/build.gradle
+++ b/android/app/build.gradle
@@ -4,24 +4,14 @@ buildscript {
     mavenCentral()
   }
   //
-  // The magic below is needed to disable Google Mobile Services (a.k.a GMS) and
-  // Google Firebase Services during the build time. Unfortunately, the only way
-  // to disable Gradle plugins is to add these hardcore switches to buildscript().
+  // The magic below is needed to disable Google Firebase Services during the build time.
+  // Unfortunately, the only way to disable Gradle plugins is to add these hardcore switches to buildscript().
   //
 
   // Detect flavors from the task name.
   def taskName = getGradle().getStartParameter().getTaskRequests().toString().toLowerCase()
   def isFdroid = taskName.contains('fdroid')
   def isBeta = taskName.contains('beta')
-
-  //
-  // Please add symlinks to google/java/app/organicmaps/location for each new gms-enabled flavor below:
-  // ```
-  // mkdir -p src/$flavor/java/app/organicmaps/
-  // ln -sf ../../../../google/java/app/organicmaps/location src/$flavor/java/app/organicmaps/
-  // ls -la src/$flavor/java/app/organicmaps/location/GoogleFusedLocationProvider.java
-  // ```
-  ext.googleMobileServicesEnabled = taskName.contains('google') || taskName.contains('huawei') || taskName.contains('web')
 
   // Firebase Crashlytics compile-time feature flag: -Pfirebase=true|false
   def googleFirebaseServicesFlag = findProperty('firebase')
@@ -34,15 +24,9 @@ buildscript {
   dependencies {
     classpath 'com.android.tools.build:gradle:8.2.1'
 
-    if (googleMobileServicesEnabled) {
-      println('Building with Google Mobile Services')
-      classpath 'com.google.gms:google-services:4.4.0'
-    } else {
-      println('Building without Google Services')
-    }
-
     if (googleFirebaseServicesEnabled) {
       println('Building with Google Firebase Services')
+      classpath 'com.google.gms:google-services:4.4.0'
       classpath 'com.google.firebase:firebase-crashlytics-gradle:2.9.9'
       classpath 'com.google.firebase:firebase-appdistribution-gradle:4.0.1'
     } else {
@@ -62,10 +46,8 @@ repositories {
 
 apply plugin: 'com.android.application'
 apply from: 'secure.properties'
-if (googleMobileServicesEnabled) {
-  apply plugin: 'com.google.gms.google-services'
-}
 if (googleFirebaseServicesEnabled) {
+  apply plugin: 'com.google.gms.google-services'
   apply plugin: 'com.google.firebase.crashlytics'
   apply plugin: 'com.google.firebase.appdistribution'
 }
@@ -372,10 +354,18 @@ android {
 }
 
 dependencies {
-  // Google Mobile Services
-  if (googleMobileServicesEnabled) {
-    implementation 'com.google.android.gms:play-services-location:21.0.1'
-  }
+  // Google Play Location Services
+  //
+  // Please add symlinks to google/java/app/organicmaps/location for each new gms-enabled flavor below:
+  // ```
+  // mkdir -p src/$flavor/java/app/organicmaps/
+  // ln -sf ../../../../google/java/app/organicmaps/location src/$flavor/java/app/organicmaps/
+  // ls -la src/$flavor/java/app/organicmaps/location/GoogleFusedLocationProvider.java
+  // ```
+  //
+  webImplementation 'com.google.android.gms:play-services-location:21.1.0'
+  googleImplementation 'com.google.android.gms:play-services-location:21.1.0'
+  huaweiImplementation 'com.google.android.gms:play-services-location:21.1.0'
 
   // Google Firebase Services
   if (googleFirebaseServicesEnabled) {

--- a/configure.sh
+++ b/configure.sh
@@ -55,7 +55,7 @@ setup_opensource() {
   echo '<?xml version="1.0" encoding="utf-8"?>
 <network-security-config/>
 ' > "$BASE_PATH/$PRIVATE_NETWORK_CONFIG"
-  echo '{"client": [{"api_key": [{"current_key": ""}], "client_info": {"mobilesdk_app_id": "omaps", "android_client_info": {"package_name": "app.organicmaps.debug"}}}], "configuration_version": "1", "project_info": {"storage_bucket": "", "project_id": "", "project_number": ""}}' > "$BASE_PATH/$PRIVATE_GOOGLE_SERVICES"
+  rm -f "$BASE_PATH/$PRIVATE_GOOGLE_SERVICES"
 }
 
 # Clone the private repo and copy all of its files (except README.md) into the main repo

